classes/decision/Workflow.inc.php

```diff
@@ -0,0 +1,125 @@
+<?php
+/**
+ * @file classes/decision/workflow.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class decision
+ *
+ * @brief A base class to define workflow steps for a particular decision type
+ */
+
+namespace PKP\decision;
+
+use APP\facades\Repo;
+use APP\submission\Submission;
+use PKP\context\Context;
+use PKP\db\DAORegistry;
+use PKP\stageAssignment\StageAssignment;
+use PKP\stageAssignment\StageAssignmentDAO;
+use PKP\submission\reviewRound\ReviewRound;
+use PKP\user\User;
+
+class Workflow
```

Comment:
I think of a workflow as any set of steps/tasks that are completed sequentially. For example an author's submission workflow, or an editor's review workflow, etc. The context information is important, though, and that's where the namespace can be important.

When I read this class I read it as "Decision Workflow" because of the namespace. I could change it to `class DecisionWorkflow`, but I guess my sense is that the namespace should provide the context of the class, in order to keep the code less verbose. Like we use `PKP\submission\Repository` instead of `PKP\submission\SubmissionRepository`.

I'm not too committed either way, though. Maybe "workflow" is just the barrier? In the UI Library I've called the components `Steps.vue` and `Step.vue`. I could mirror that and call this `class PKP\decision\Steps`. Would that be better?

Reply:
I have a very slight preference for steps, especially since we're already using it elsewhere. I'm probably reacting to "workflow" because we've had so many discussions about "flexible workflow" and it carries those connotations. So if you think it's worth adapting, please go ahead -- but if you're really OK with it as it stands, then I am too.

Reply:
Sounds good, I've changed it to `Steps`.